### PR TITLE
Sign arm binaries in pipeline

### PR DESCRIPTION
**Pull request by wtgodbe**
Dup of https://github.com/dotnet/core-setup/pull/2311, attempts to resolve https://github.com/dotnet/core-setup/issues/1977.

I ran a build with these changes by merging https://github.com/dotnet/core-setup/pull/2311 (I have since reverted that merge, hence this PR), which produced these results: https://devdiv.visualstudio.com/DevDiv/_build/explorer?_a=summary&buildId=737498. The only job that passed was the arm job, oddly (arm64 failed). @dagood @eerhardt @chcosta @gkhanna79 any idea why the non-Windows jobs would fail, or what would have caused the test failures in the x64/x86 jobs?

### CODE REVIEW

**Comment by gkhanna79** [comment]
A bunch of legs failed with the following:

```
2017-05-12T19:53:25.7357290Z chmod: cannot access '/home/DDITAdministrator/myagent/_work/161/a/HostSandbox/packages/microsoft.dotnet.buildtools/2.0.0-prerelease-01611-07/lib/init-tools.sh': No such file or directory
2017-05-12T19:53:25.7385470Z /home/DDITAdministrator/myagent/_work/161/a/HostSandbox/init-tools.sh: line 102: /home/DDITAdministrator/myagent/_work/161/a/HostSandbox/packages/microsoft.dotnet.buildtools/2.0.0-prerelease-01611-07/lib/init-tools.sh: No such file or directory
2017-05-12T19:53:25.7418080Z ERROR: An error occured when trying to initialize the tools. Please check '/home/DDITAdministrator/myagent/_work/161/a/HostSandbox/init-tools.log' for more details.1
2017-05-12T19:53:25.7568190Z ##[error]/home/DDITAdministrator/myagent/_work/161/a/HostSandbox/init-tools.sh failed with return code: 1
2017-05-12T19:53:25.7641670Z ##[error]/home/DDITAdministrator/myagent/_work/161/a/HostSandbox/init-tools.sh failed with error: /home/DDITAdministrator/myagent/_work/161/a/HostSandbox/init-tools.sh failed with return code: 1
2017-05-12T19:53:25.8286140Z ##[section]Finishing: Initialize tools in sandbox for host machine
```

@markwilkie @MattGal Do you know anything about this?

**Comment by gkhanna79** [comment]
```
2017-05-12T19:54:39.0577413Z ##[error]EXEC(0,0): Error : DIA SDK is missing at "DIA SDK". This is due to a bug in the Visual Studio installer. It does not install DIA SDK at "" but rather at VS install location of previous version. Workaround is to copy DIA SDK folder from VS install location of previous version to "" and then resume build.
2017-05-12T19:54:39.0577413Z EXEC : error : DIA SDK is missing at "DIA SDK". This is due to a bug in the Visual Studio installer. It does not install DIA SDK at "" but rather at VS install location of previous version. Workaround is to copy DIA SDK folder from VS install location of previous version to "" and then resume build. [E:\A\_work\145\s\core-setup\src\corehost\build.proj]
2017-05-12T19:54:39.0577413Z   Visual Studio 2013 Express does not include the DIA SDK. You need Visual Studio 2013+ (Community is free).
2017-05-12T19:54:39.0577413Z   See: https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/developer-guide.md#prerequisites
2017-05-12T19:54:39.0577413Z ##[error]core-setup\src\corehost\build.proj(60,5): Error MSB3073: The command "E:\A\_work\145\s\core-setup\src\corehost\build.cmd Release arm64 apphostver 2.0.0-preview2-25312-02 hostver 2.0.0-preview2-25312-02 fxrver 2.0.0-preview2-25312-02 policyver 2.0.0-preview2-25312-02 commit 06e6d0e75d5d612e98a99f702a4206a75c515c6b rid win-arm64 portable toolsetdir C:\tools\clr" exited with code 1.
2017-05-12T19:54:39.0577413Z E:\A\_work\145\s\core-setup\src\corehost\build.proj(60,5): error MSB3073: The command "E:\A\_work\145\s\core-setup\src\corehost\build.cmd Release arm64 apphostver 2.0.0-preview2-25312-02 hostver 2.0.0-preview2-25312-02 fxrver 2.0.0-preview2-25312-02 policyver 2.0.0-preview2-25312-02 commit 06e6d0e75d5d612e98a99f702a4206a75c515c6b rid win-arm64 portable toolsetdir C:\tools\clr" exited with code 1.
2017-05-12T19:54:39.0577413Z Done Building Project "E:\A\_work\145\s\core-setup\src\corehost\build.proj" (Build target(s)) -- FAILED.
2017-05-12T19:54:39.0577413Z Done Building Project "E:\A\_work\145\s\core-setup\src\src.builds" (default targets) -- FAILED.
2017-05-12T19:54:39.0577413Z 
2017-05-12T19:54:39.0577413Z Build FAILED.
2017-05-12T19:54:39.0577413Z 
2017-05-12T19:54:39.0577413Z "E:\A\_work\145\s\core-setup\src\src.builds" (default target) (1) ->
```

Win-Arm leg failed due to this - @ravimeda Are you aware of any toolset changes for CoreSetup master?

CC @weshaggard

**Comment by gkhanna79** [comment]
```
2017-05-12T20:07:27.7776220Z   Failed   Microsoft.DotNet.CoreSetup.Test.HostActivation.PortableApp.GivenThatICareAboutPortableAppActivation.Muxer_Exec_activation_of_Build_Output_Portable_DLL_with_DepsJson_Local_and_RuntimeConfig_Remote_With_AdditionalProbingPath_Succeeds
2017-05-12T20:07:27.7806170Z   Error Message:
2017-05-12T20:07:27.7833990Z    Expected command to pass but it did not.
2017-05-12T20:07:27.7867260Z   File Name: /Users/buildagent/agent/_work/15/s/core-setup/Bin/obj/osx-x64.Release/sharedFrameworkPublish/dotnet
2017-05-12T20:07:27.7903890Z   Arguments: exec --runtimeconfig /Users/buildagent/agent/_work/15/s/core-setup/Bin/tests/osx-x64.Release/12/PortableApp/r/PortableApp.runtimeconfig.json --additionalprobingpath /Users/buildagent/agent/_work/15/s/core-setup/packages /Users/buildagent/agent/_work/15/s/core-setup/Bin/tests/osx-x64.Release/12/PortableApp/bin/PortableApp.dll
2017-05-12T20:07:27.7932340Z   Exit Code: 140
2017-05-12T20:07:27.7945840Z   StdOut:
2017-05-12T20:07:27.7960220Z   
2017-05-12T20:07:27.7973470Z   StdErr:
2017-05-12T20:07:27.7989620Z   Error: assembly specified in the dependencies manifest was not found -- package: 'Newtonsoft.Json', version: '9.0.1', path: 'lib/netstandard1.0/Newtonsoft.Json.dll'
2017-05-12T20:07:27.8007130Z   
2017-05-12T20:07:27.8022380Z   
2017-05-12T20:07:27.8036420Z   Stack Trace:
```

Three Windows legs (win-arm, win-x64, etc) failed with this. 

I am trying OSX build to see what is going on. Surprisingly, we had a good build earlier today (https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=737210&_a=summary).

CC @eerhardt

**Comment by gkhanna79** [comment]
Between the passing commit (https://github.com/dotnet/core-setup/commit/a11337e901fcea4cad5fc7990ffc2efb278f2f71) and commit corresponding to first failing build (https://github.com/dotnet/core-setup/commit/06e6d0e75d5d612e98a99f702a4206a75c515c6b), https://github.com/dotnet/core-setup/pull/2373 seems to be suspect commit for test failures.

@mellinoe Can you PTAL?

File: buildpipeline/pipeline.json [hunk 1]
```diff
@@ -112,7 +111,7 @@
         {
           "Name": "Core-Setup-OSX-BT",
           "Parameters": {
-            "PB_PortableBuildArg": "-PortableBuild=true",
+            "PB_PortableBuildArg": "-portable",
```
Comment (wtgodbe) [inline buildpipeline/pipeline.json]: @gkhanna79 @eerhardt should this change not have been made?

Comment (gkhanna79) [inline buildpipeline/pipeline.json]: Yes, this is incorrect after https://github.com/dotnet/core-setup/commit/64a0b92cb524427a620f9760a4f02b41444eb089 and should be reverted.

File: buildpipeline/pipeline.json [hunk 2]
```diff
@@ -82,7 +82,7 @@
           "Parameters": {
             "PB_DistroRid": "rhel.7.2-x64",
             "PB_DockerTag": "rhel7_prereqs_2",
-            "PB_AdditionalBuildArguments":"-PortableBuild=true",
+            "PB_AdditionalBuildArguments":"-portable",
```
Comment (gkhanna79) [inline buildpipeline/pipeline.json]: This is incorrect as well.

File: buildpipeline/pipeline.json [hunk 3]
```diff
@@ -98,8 +98,7 @@
             "PB_DistroRid": "ubuntu.14.04-arm", 
             "PB_DockerTag": "ubuntu-14.04-cross-0cd4667-20172211042239", 
             "PB_TargetArchitecture": "arm", 
-            "PB_AdditionalBuildArguments":"-TargetArchitecture=arm -DistroRid=linux-arm -DisableCrossgen=true -PortableBuild=true -SkipTests=true -CrossBuild=true",
-            "PB_CrossBuildArgs": "-e ROOTFS_DIR ",
+            "PB_AdditionalBuildArguments":"arm cross disablecrossgen -portable skiptests",
```
Comment (gkhanna79) [inline buildpipeline/pipeline.json]: This should be reverted too.

File: buildpipeline/pipeline.json [hunk 4]
```diff
@@ -98,8 +98,7 @@
             "PB_DistroRid": "ubuntu.14.04-arm", 
             "PB_DockerTag": "ubuntu-14.04-cross-0cd4667-20172211042239", 
             "PB_TargetArchitecture": "arm", 
-            "PB_AdditionalBuildArguments":"-TargetArchitecture=arm -DistroRid=linux-arm -DisableCrossgen=true -PortableBuild=true -SkipTests=true -CrossBuild=true",
-            "PB_CrossBuildArgs": "-e ROOTFS_DIR ",
```
Comment (gkhanna79) [inline buildpipeline/pipeline.json]: Revert this please.

**Comment by wtgodbe** [comment]
@gkhanna79 I updated pipelines.json, does it look good now?

**Comment by mellinoe** [comment]
> Error: assembly specified in the dependencies manifest was not found -- package: 'Newtonsoft.Json', version: '9.0.1', path: 'lib/netstandard1.0/Newtonsoft.Json.dll'

This can happen when the first-time experience for the CLI has run, because it populates the "fallback folder" with version 9.0.1 of Newtonsoft.Json. When the tests run, they don't restore that package because it is in the fallback folder. It's possible that the build just ran on a machine that was broken in that way, but will succeed on another machine. cc: @eerhardt

**Comment by gkhanna79** [comment]
@mellinoe Just so I get it, are you saying that the fallback folder initialization failed on this machine?

File: buildpipeline/pipeline.json [hunk 5]
```diff
@@ -98,7 +98,7 @@
             "PB_DistroRid": "ubuntu.14.04-arm", 
             "PB_DockerTag": "ubuntu-14.04-cross-0cd4667-20172211042239", 
             "PB_TargetArchitecture": "arm", 
-            "PB_AdditionalBuildArguments":"-TargetArchitecture=arm -DistroRid=linux-arm -DisableCrossgen=true -PortableBuild=true -SkipTests=true -CrossBuild=true",
+            "PB_AdditionalBuildArguments":"-TargetArchitecture=arm -DistroRid=linux-arm -DisableCrossgen=true -PortableBuild=true -SkipTests=true",
```
Comment (gkhanna79) [inline buildpipeline/pipeline.json]: Why was -CrossBuild=true removed from here?

Comment (wtgodbe) [inline buildpipeline/pipeline.json]: I duplicated the change from the commit you referenced - https://github.com/dotnet/core-setup/commit/64a0b92cb524427a620f9760a4f02b41444eb089. Should I add -CrossBuild=true back in?

Comment (gkhanna79) [inline buildpipeline/pipeline.json]: What I said was that commit introduced the new way of specifying the arguments. My PR, post that commit, added -Crossbuild=true. 

So, yes, we should revert this change.

Comment (wtgodbe) [inline buildpipeline/pipeline.json]: Gotcha. Will do that now.

**Comment by mellinoe** [comment]
@gkhanna79 I'm saying the fallback folder was successfully initialized -- that causes the tests to fail. This is just based on my discussions with @eerhardt while working through test failures.

**Comment by gkhanna79** [comment]
> When the tests run, they don't restore that package because it is in the fallback folder.

Do tests restore packages in a location different from fallback folder? If not, then should they be passed the path to the fallback folder so that the dependency is located?

**Comment by mellinoe** [comment]
> Do tests restore packages in a location different from fallback folder?

I'm not very familiar with the fallback folder feature, but I don't think anybody restores packages to it, other than the initialization step. It's a different folder from the regular nuget package cache (~/.nuget/packages).

**Comment by gkhanna79** [comment]
@wtgodbe What are the next steps for this?
